### PR TITLE
Fix TTS IsReady false issues

### PR DESCRIPTION
**Pull request by OHerasym**
### Pull request changes:
- Fix TTS IsReady false changes

Related: APPLINK-28576, 28577, 28662

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @AGritsevich , @vlantonov , @VProdanov , @anosach-luxoft , @orhan-mehmedov, @dtrunov

File: src/components/application_manager/src/commands/mobile/change_registration_request.cc [hunk 1]
```diff
@@ -318,6 +325,11 @@ bool ChangeRegistrationRequest::PrepareResponseParameters(
     result_code = MessageHelper::HMIToMobileResult(
         std::max(std::max(ui_result, vr_result), tts_result));
   }
+
+  CheckInfo(ui_response_info_);
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/change_registration_request.cc]: @OHerasym Wrrong logic please discuss with @AGritsevich

File: src/components/application_manager/src/commands/mobile/change_registration_request.cc [hunk 2]
```diff
@@ -285,6 +284,13 @@ bool ChangeRegistrationRequest::PrepareResponseParameters(
           ui_result_,
           vr_result_);
 
+  const HmiInterfaces& hmi_interfaces = application_manager_.hmi_interfaces();
+  const HmiInterfaces::InterfaceState tts_state =
+      hmi_interfaces.GetInterfaceState(
+          HmiInterfaces::InterfaceID::HMI_INTERFACE_TTS);
+  const bool tts_state_is_generated =
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/change_registration_request.cc]: @OHerasym Maybe change name to is_tts_state_not_available

File: src/components/application_manager/src/commands/mobile/change_registration_request.cc [hunk 3]
```diff
@@ -295,8 +301,9 @@ bool ChangeRegistrationRequest::PrepareResponseParameters(
           ui_result_,
           vr_result_);
 
-  if ((result && is_tts_or_ui_or_vr_unsupported) || is_tts_ui_vr_unsupported) {
+  if ((result && is_tts_or_ui_or_vr_unsupported)) {
     result_code = mobile_apis::Result::UNSUPPORTED_RESOURCE;
+    result = tts_state_is_generated & !is_tts_or_ui_or_vr_unsupported;
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/change_registration_request.cc]: @OHerasym Maybe && instead &? Please remake "result = tts_state_is_generated & !is_tts_or_ui_or_vr_unsupported";

**Pull request merged**